apps/nextra/pages/en/build/cli/use-cli/working-with-move-contracts/local-simulation-benchmarking-and-gas-profiling.mdx

```diff
@@ -0,0 +1,251 @@
+- `--local`: Simulate the transaction locally without additional alterations.
+- `--benchmark`: Benchmarck the transaction and report the running time(s).
+- `--profile-gas`: Profile the transaction for detailed gas usage.
+
+These additional options can be used in combination with the following CLI commands:
+- `aptos move run`
+- `aptos move run-script`
+- `aptos move publish`
+
+<Callout type="info" emoji="ℹ️">
+Local simulations do not result in any changes of the chain state.
+</Callout>
+
+## Deploying the Example Contract
+For demonstration purposes, we will continue to use the [`hello_blockchain`](https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/move-examples/hello_blockchain) package as an example.
+
+First, publish the package to devnet or testnet (if you haven't done so already).
+
+Change into the package directory.
+```bash filename="Terminal"
+cd aptos-move/move-examples/hello_blockchain
+```
+
+Then publish the package using the following command.
+```bash filename="Terminal"
+aptos move publish --named-addresses hello_blockchain=default --assume-yes
+```
+
+<details>
+<summary>Output</summary>
+```bash
+{
+  "Result": {
+    "transaction_hash": "0xe4ae0ec4ea3474b2123838885b04d7f4b046c174d14d7dc1c56916f2eb553bcf",
+    "gas_used": 1118,
+    "gas_unit_price": 100,
+    "sender": "dbcbe741d003a7369d87ec8717afb5df425977106497052f96f4e236372f7dd5",
+    "sequence_number": 5,
+    "success": true,
+    "timestamp_us": 1713914742422749,
+    "version": 1033819503,
+    "vm_status": "Executed successfully"
+  }
+}
+```
+</details>
+
+Notice that you do need to have your CLI profile set up properly and bind the named addresses correctly. Please refer to [CLI Configuration](../../cli-configuration.md) for more details.
+
+
+## Local Simulation
+Next, execute the entry function message::set_message with local simulation enabled using the additional command line option `--local`. This will execute the transaction locally without conducting any further measurements or analysis.
+```bash filename="Terminal"
+aptos move run --function-id 'default::message::set_message' --args 'string:abc' --local
+```
+
+<details>
+<summary>Output</summary>
+```bash
+Simulating transaction locally...
+{
+  "Result": {
+    "transaction_hash": "0x5aab20980688185eed2c9a27bab624c84b8b8117241cd4a367ba2a012069f57b",
+    "gas_used": 441,
+    "gas_unit_price": 100,
+    "sender": "dbcbe741d003a7369d87ec8717afb5df425977106497052f96f4e236372f7dd5",
+    "success": true,
+    "version": 1033887414,
+    "vm_status": "status EXECUTED of type Execution"
+  }
+}
+```
+</details>
+<Callout type="info" emoji="ℹ️">
+Local and remote simulation shall produce identical results.
+</Callout>
+
+## Benchmarking
+To measure the running time(s) of your transaction, use the `--benchmark` option.
+
+```bash filename="Terminal"
+aptos move run --function-id 'default::message::set_message' --args 'string:abc' --benchmark
```

Comment:
It would be nice to mention that we can replay and benchmark / profile gas any transaction at version X - I think it is a great feature. Not sure if in this doc because it talks about local simulations, but we should mention that 100%

Reply:
There will be a dedicated PR for the new replay command. It will come in a subsequent PR!